awx/main/tasks/jobs.py

```diff
@@ -400,6 +400,11 @@ def run(self, pk, **kwargs):
         Run the job/task and capture its output.
         """
         self.instance = self.model.objects.get(pk=pk)
+        if self.instance.status != 'canceled' and self.instance.cancel_flag:
+            self.instance = self.update_model(self.instance.pk, start_args='', status='canceled')
+        if self.instance.status not in ACTIVE_STATES:
+            # Prevent starting the job if it has been reaped or handled by another process.
+            raise RuntimeError(f'Not starting {self.instance.status} task pk={pk}')
```

Comment:
maybe include 
```suggestion
            raise RuntimeError(f'Not starting {self.instance.status} task pk={pk} because {self.instance.status} is not in a valid active state')
```